testing/asserts.ts

```diff
@@ -541,7 +541,17 @@ export function assertObjectMatch(
       if ((seen.has(a)) && (seen.get(a) === b)) {
         return a;
       }
-      seen.set(a, b);
+      try {
+        seen.set(a, b);
+      } catch (err) {
+        if (err instanceof TypeError) {
+          throw new TypeError(
+            `Cannot assertObjectMatch ${
+              a === null ? null : `type ${typeof a}`
+            }`,
+          );
```

Comment:
How about returning `a` here instead of throwing `TypeError`? That makes the case `assertObjectMatch(null, { foo: 42 })` throwing more useful AssertionError instead of TypeError.

Reply:
I'm open to that. I feel like assertObjectMatch is supposed to be for higher ordered objects, so I've been wanting to do a typeerror, but I understand you could have a function that returns either null/undefined or an object. The problem I have is there are ts warnings whenever you try to pass something that couldn't be made into a weakmap, so if we go that route, I wonder if we update the argument types to accept more values than 
```
actual: Record<PropertyKey, any>,
expected: Record<PropertyKey, unknown>,
```
I'm pushing the updates so we can see how it looks.

Reply:
> I feel like assertObjectMatch is supposed to be for higher ordered objects, so I've been wanting to do a typeerror

Ah, right. The previous version might make more sense. Thanks for the clarification of the reasoning.